lib/HLSL/DxilContainerReflection.cpp

```diff
@@ -180,16 +182,14 @@ class DxilShaderReflection : public DxilModuleReflection, public ID3D12ShaderRef
   DXC_MICROCOM_TM_ADDREF_RELEASE_IMPL()
   DXC_MICROCOM_TM_CTOR(DxilShaderReflection)
   HRESULT STDMETHODCALLTYPE QueryInterface(REFIID iid, void **ppvObject) override {
-    HRESULT hr = DoBasicQueryInterface<ID3D12ShaderReflection>(this, iid, ppvObject);
```

Comment:
Why is it ok to completely drop the DoBasicQueryInterface call here?

Reply:
Because we will manually check against `IUnknown` and the IID of the interface we are supposed to be emulating only.

Before, it would say if you QI for `ID3D12ShaderReflection`, it would be ok, even though our `m_PublicAPI` might be a different version.  When you call a method on the interface, you could get behavior that's incorrect for `ID3D12ShaderReflection`, because we use `m_PublicAPI` to determine certain behaviors later.